grpc/wrappers.go

```diff
@@ -583,3 +558,865 @@ func (ras *RegistrationAuthorityServerWrapper) AdministrativelyRevokeCertificate
+	if err != nil {
+		return core.Registration{}, unwrapError(err)
+	}
+
+	if response == nil || !registrationValid(response) {
+		return core.Registration{}, errIncompleteResponse
+	}
+
+	return pbToRegistration(response)
+}
+
+func (sac StorageAuthorityClientWrapper) GetAuthorization(ctx context.Context, authID string) (core.Authorization, error) {
+	localCtx, cancel := context.WithTimeout(ctx, sac.timeout)
+	defer cancel()
+
+	response, err := sac.inner.GetAuthorization(localCtx, &sapb.AuthorizationID{Id: &authID})
+	if err != nil {
+		return core.Authorization{}, unwrapError(err)
+	}
+
+	if response == nil || !authorizationValid(response) {
+		return core.Authorization{}, errIncompleteResponse
+	}
+
+	return pbToAuthz(response)
+}
+
+func (sac StorageAuthorityClientWrapper) GetValidAuthorizations(ctx context.Context, regID int64, domains []string, now time.Time) (map[string]*core.Authorization, error) {
+	localCtx, cancel := context.WithTimeout(ctx, sac.timeout)
+	defer cancel()
+
+	nowUnix := now.UnixNano()
+
+	response, err := sac.inner.GetValidAuthorizations(localCtx, &sapb.GetValidAuthorizationsRequest{
+		RegistrationID: &regID,
+		Domains:        domains,
+		Now:            &nowUnix,
+	})
+	if err != nil {
+		return nil, unwrapError(err)
+	}
+
+	if response == nil {
+		return nil, errIncompleteResponse
+	}
+
+	auths := make(map[string]*core.Authorization, len(response.Valid))
+	for _, element := range response.Valid {
+		if element == nil || element.Domain == nil || !authorizationValid(element.Authz) {
+			return nil, errIncompleteResponse
+		}
+		authz, err := pbToAuthz(element.Authz)
+		if err != nil {
+			return nil, err
+		}
+		auths[*element.Domain] = &authz
+	}
+	return auths, nil
+}
+
+func (sac StorageAuthorityClientWrapper) GetCertificate(ctx context.Context, serial string) (core.Certificate, error) {
+	localCtx, cancel := context.WithTimeout(ctx, sac.timeout)
+	defer cancel()
+
+	response, err := sac.inner.GetCertificate(localCtx, &sapb.Serial{Serial: &serial})
+	if err != nil {
+		return core.Certificate{}, unwrapError(err)
+	}
+
+	if response == nil || !certificateValid(response) {
+		return core.Certificate{}, errIncompleteResponse
+	}
+
+	return pbToCert(response), nil
+}
+
+func (sac StorageAuthorityClientWrapper) GetCertificateStatus(ctx context.Context, serial string) (core.CertificateStatus, error) {
+	localCtx, cancel := context.WithTimeout(ctx, sac.timeout)
+	defer cancel()
+
+	response, err := sac.inner.GetCertificateStatus(localCtx, &sapb.Serial{Serial: &serial})
+	if err != nil {
+		return core.CertificateStatus{}, unwrapError(err)
+	}
+
+	if response == nil || response.Serial == nil || response.SubscriberApproved == nil || response.Status == nil || response.OcspLastUpdated == nil || response.RevokedDate == nil || response.RevokedReason == nil || response.LastExpirationNagSent == nil || response.OcspResponse == nil || response.NotAfter == nil || response.IsExpired == nil {
```

Comment:
Since we only check `sapb.CertificateStatus` once I didn't think it required its own method.